### PR TITLE
Fix typos throughout the source code

### PR DESCRIPTION
**Pull request by NeeEoo**
There were some inconsistencies with British and American english, but I'm not gonna fix that, since its technically correct, and I don't know what language the documentation is supposed to be.

### CODE REVIEW

**Comment by NeeEoo** [comment]
There were also some stuff I noticed,
cpp spells it finalize
and eval spells it finalise

File: std/haxe/macro/Compiler.hx [hunk 1]
```diff
@@ -397,7 +397,7 @@ class Compiler {
 	}
 
 	/**
-		Register a custom medatada for documentation and completion purposes
```
Comment (Simn) [inline std/haxe/macro/Compiler.hx:400]: lol metatada...

**Comment by Simn** [comment]
Thanks! It's a bit scary how many typos there were...

File: std/haxe/macro/Compiler.hx (hunk 1 above)
Comment (kLabz) [reply std/haxe/macro/Compiler.hx:400]: My hands keep making that one, have to type it 2 or 3 times to get it right x_x

**Pull request merged**